geth/transactions/txqueue_manager.go

```diff
@@ -83,127 +68,100 @@ func (m *Manager) TransactionQueue() common.TxQueue {
 	return m.txQueue
 }
 
-// CreateTransaction returns a transaction object.
-func (m *Manager) CreateTransaction(ctx context.Context, args common.SendTxArgs) *common.QueuedTx {
-	return &common.QueuedTx{
-		ID:      common.QueuedTxID(uuid.New()),
-		Hash:    gethcommon.Hash{},
-		Context: ctx,
-		Args:    args,
-		Done:    make(chan struct{}, 1),
-		Discard: make(chan struct{}, 1),
-	}
-}
-
 // QueueTransaction puts a transaction into the queue.
 func (m *Manager) QueueTransaction(tx *common.QueuedTx) error {
 	to := "<nil>"
 	if tx.Args.To != nil {
 		to = tx.Args.To.Hex()
 	}
 	log.Info("queue a new transaction", "id", tx.ID, "from", tx.Args.From.Hex(), "to", to)
+	err := m.txQueue.Enqueue(tx)
+	if m.notify {
```

Comment:
It could be beneficial to create a method
```go
func (m *Manager) notifyOnEnqueue(tx) {
  if m.notify {
    NotifyOnEnqueue(tx)
  }
}
```
so that the whole logic when to send a notification is encapsulated in one place. Here, if there will be another condition, all `NotifyOnEnqueue` call conditions would need to be changed.